File: app/src/main/java/ru/vladislav_akulinin/mychat_version_2/StartActivity.java
package ru.vladislav_akulinin.mychat_version_2;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class StartActivity extends AppCompatActivity {

    Button register, login;

    FirebaseUser firebaseUser;

    @Override
    protected void onStart() {
        super.onStart();

        //инициализируем объект и получаем текущего пользователя (чтобы не авторизовываться постоянно)
        firebaseUser = FirebaseAuth.getInstance().getCurrentUser(); //получаем текущего пользователя, который вошел в систему

        //если пользователь уже вошел в систему, то переправляем его на новую активность (чтобы не авторизовываться постоянно)
        if(firebaseUser != null){
            Intent intent = new Intent(StartActivity.this, MainActivity.class);
            startActivity(intent);
            finish();
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start);

        login = findViewById(R.id.login);
        register = findViewById(R.id.register);

        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(StartActivity.this, LoginActivity.class));
            }
        });

        register.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(StartActivity.this, RegisterActivity.class));
            }
        });
    }
}

File: app/src/main/java/ru/vladislav_akulinin/mychat_version_2/ui/activity/StartActivity.kt
package ru.vladislav_akulinin.mychat_version_2.ui.activity

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import ru.vladislav_akulinin.mychat_version_2.R

class StartActivity : AppCompatActivity() {

    private var firebaseUser: FirebaseUser ?= null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_start)

        findViewById<Button>(R.id.login).setOnClickListener{
            startActivity(Intent(this@StartActivity, LoginActivity::class.java))
        }

        findViewById<Button>(R.id.register).setOnClickListener{
            startActivity(Intent(this@StartActivity, RegisterActivity::class.java))
        }
    }

    override fun onStart() {
        super.onStart()

        //получаем текущего пользователя, который вошел в систему
        firebaseUser = FirebaseAuth.getInstance().currentUser

        //если пользователь уже вошел в систему, то переправляем его на новую активность (чтобы не авторизовываться постоянно)
        if (firebaseUser != null) {
            val intent = Intent(this@StartActivity, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }
}